src/gui/sdlmain.cpp

```diff
@@ -3850,6 +3842,22 @@ bool GFX_Events()
 			continue;
 		}
 		switch (event.type) {
+		case SDL_JOYDEVICEADDED: {
+			SDL_JoyDeviceEvent* added_event =
+			        reinterpret_cast<SDL_JoyDeviceEvent*>(&event);
+			LOG_WARNING("MAPPER: Joystick device was just added, index: '%d'",
+			            added_event->which);
+			MAPPER_BindKeys(get_sdl_section());
+			break;
+		}
+		case SDL_JOYDEVICEREMOVED: {
+			SDL_JoyDeviceEvent* removed_event =
+			        reinterpret_cast<SDL_JoyDeviceEvent*>(&event);
+			LOG_WARNING("MAPPER: Joystick device was just removed, instance ID: '%d'",
+			            removed_event->which);
+			MAPPER_BindKeys(get_sdl_section());
+			break;
```

Comment:
Comments about both the SDL_JOYDEVICEADDED and SDL_JOYDEVICEREMOVED log messages:

- Should be info level, not warn.
- Should be ", instance ID %d" for both, but these instances IDs tell absolutely nothing to the user. Would it be possible to look up the human-readable name of the device as it reports itself to the OS and log that name instead?
- I'm not quite sure how the differentiation between "joysticks" and "game controllers" work in SDL. Don't we need to handle both (see [this event](https://github.com/libsdl-org/SDL/blob/main/include/SDL3/SDL_events.h#L645-L665)? In your tests you were using Xbox game controllers, so I'm a bit confused why these report themselves as "joysticks", not "game controllers". What about the extra buttons on game controllers like the Xbox 360; those are fully mappable in Staging. So wouldn't it be better if we only dealt with "game controllers" conceptually?
- So, I suggest to log something like this instead: `MAPPER: Game controller Xbox 360 has been plugged in." (or whatever the human-readable name of the controller is). And then `MAPPER: Game controller Xbox 360 has been disconnected."